crates/sui/src/ethereum_client_commands.rs

```diff
@@ -239,3 +229,289 @@ pub(crate) async fn create_eth_dwallet(
+    }
+
+    let data_slot = latest_eth_state_obj.eth_smart_contract_slot;
+    let contract_addr: String = latest_eth_state_obj.eth_smart_contract_address;
+    let contract_addr = contract_addr.clone().parse::<Address>()?;
+
+    let mut eth_lc = EthLightClientWrapper::init_new_light_client(eth_lc_config.clone()).await?;
+
+    let updates_response = eth_state
+        .get_updates_since_checkpoint()
+        .await
+        .map_err(|e| anyhow!("could not fetch updates: {e}"))?;
+
+    let gas_owner = context.try_get_object_owner(&gas).await?;
+    let sender = gas_owner.unwrap_or(context.active_address()?);
+
+    // Serialize Move parameters
+    let mut pt_builder = ProgrammableTransactionBuilder::new();
+    let eth_state_arg = pt_builder
+        .pure(serde_json::to_vec(&eth_state)?)
+        .map_err(|e| anyhow!("could not serialize eth_state. {e}"))?;
+
+    let updates_vec_arg = pt_builder
+        .pure(serde_json::to_vec(&updates_response.updates)?)
+        .map_err(|e| anyhow!("could not serialize updates. {e}"))?;
+
+    let finality_update_arg = pt_builder
+        .pure(serde_json::to_vec(&updates_response.finality_update)?)
+        .map_err(|e| anyhow!("could not serialize finality_updates. {e}"))?;
+
+    let optimistic_update_arg = pt_builder
+        .pure(serde_json::to_vec(&updates_response.optimistic_update)?)
+        .map_err(|e| anyhow!("could not serialize optimistic_updates. {e}"))?;
+
+    let latest_eth_state_shared_object_arg = ObjectArg::SharedObject {
+        id: latest_eth_state_shared_object.id,
+        initial_shared_version: latest_eth_state_shared_object.initial_shared_version,
+        mutable: true,
+    };
+
+    let latest_eth_state_arg = pt_builder
+        .obj(latest_eth_state_shared_object_arg)
+        .map_err(|e| anyhow!("could not serialize latest_eth_state_id. {e}"))?;
+
+    pt_builder.programmable_move_call(
+        SUI_SYSTEM_PACKAGE_ID,
+        ETHEREUM_STATE_MODULE_NAME.into(),
+        VERIFY_ETH_STATE_FUNC_NAME.into(),
+        vec![],
+        Vec::from([
+            updates_vec_arg,
+            finality_update_arg,
+            optimistic_update_arg,
+            eth_state_arg,
+            latest_eth_state_arg,
+        ]),
+    );
+
+    let client = context.get_client().await?;
+    let tx_data = client
+        .transaction_builder()
+        .finish_programmable_transaction(sender, pt_builder, gas, gas_budget)
+        .await?;
+
+    let verify_state_session_response = serialize_or_execute!(
+        tx_data,
+        serialize_unsigned_transaction,
+        serialize_signed_transaction,
+        context,
+        Call
+    );
+
+    let SuiClientCommandResult::Call(state) = verify_state_session_response else {
+        return Err(anyhow!("can't get response."));
+    };
+
+    let object_changes = state
+        .object_changes
+        .clone()
+        .ok_or_else(|| anyhow!("can't get object changes."))?;
```

Comment:
```suggestion
        .ok_or_else(|| anyhow!("can't get object changes"))?;
```
also make this error more meaningful